diff --git a/ogr/ogrsf_frmts/arrow_common/ograrrowlayer.hpp b/ogr/ogrsf_frmts/arrow_common/ograrrowlayer.hpp
index 32c2e86d9624..7810a812c6ff 100644
--- a/ogr/ogrsf_frmts/arrow_common/ograrrowlayer.hpp
+++ b/ogr/ogrsf_frmts/arrow_common/ograrrowlayer.hpp
@@ -428,6 +428,12 @@ inline bool OGRArrowLayer::MapArrowTypeToOGR(
         case arrow::Type::INTERVAL_MONTH_DAY_NANO:
 #if ARROW_VERSION_MAJOR >= 12
         case arrow::Type::RUN_END_ENCODED:
+#endif
+#if ARROW_VERSION_MAJOR >= 15
+        case arrow::Type::STRING_VIEW:
+        case arrow::Type::BINARY_VIEW:
+        case arrow::Type::LIST_VIEW:
+        case arrow::Type::LARGE_LIST_VIEW:
 #endif
         case arrow::Type::MAX_ID:
         {
@@ -2076,6 +2082,12 @@ inline OGRFeature *OGRArrowLayer::ReadFeature(
             case arrow::Type::INTERVAL_MONTH_DAY_NANO:
 #if ARROW_VERSION_MAJOR >= 12
             case arrow::Type::RUN_END_ENCODED:
+#endif
+#if ARROW_VERSION_MAJOR >= 15
+            case arrow::Type::STRING_VIEW:
+            case arrow::Type::BINARY_VIEW:
+            case arrow::Type::LIST_VIEW:
+            case arrow::Type::LARGE_LIST_VIEW:
 #endif
             case arrow::Type::MAX_ID:
             {
