--- gdal-3.0.1.orig/swig/python/setup.py	2019-06-28 07:54:28.000000000 -0300
+++ gdal-3.0.1/swig/python/setup.py	2019-10-23 11:42:51.243865345 -0300
@@ -268,17 +268,17 @@
                     if ext.name != 'osgeo._gdalconst':
                         ext.extra_compile_args += [cxx11_flag]
 
-        # Adding arch flags here if OS X and compiler is clang
-        if sys.platform == 'darwin' and [int(x) for x in os.uname()[2].split('.')] >= [11, 0, 0]:
-            # since MacOS X 10.9, clang no longer accepts -mno-fused-madd
-            # extra_compile_args.append('-Qunused-arguments')
-            clang_flag = '-Wno-error=unused-command-line-argument-hard-error-in-future'
-            if has_flag(self.compiler, clang_flag): 
-                ext.extra_compile_args += [clang_flag]
-            else:
-                clang_flag = '-Wno-error=unused-command-line-argument'
-                if has_flag(self.compiler, clang_flag):
-                    ext.extra_compile_args += [clang_flag]
+                    # Adding arch flags here if OS X and compiler is clang
+                    if sys.platform == 'darwin' and [int(x) for x in os.uname()[2].split('.')] >= [11, 0, 0]:
+                        # since MacOS X 10.9, clang no longer accepts -mno-fused-madd
+                        # extra_compile_args.append('-Qunused-arguments')
+                        clang_flag = '-Wno-error=unused-command-line-argument-hard-error-in-future'
+                        if has_flag(self.compiler, clang_flag):
+                            ext.extra_compile_args += [clang_flag]
+                        else:
+                            clang_flag = '-Wno-error=unused-command-line-argument'
+                            if has_flag(self.compiler, clang_flag):
+                                ext.extra_compile_args += [clang_flag]
 
         build_ext.build_extensions(self)
 
